app/models/user.rb

```diff
@@ -192,6 +192,30 @@ def most_accurate_school_country
     school.present? ? school.country : 'United States'
   end
 
+  def school_location
+    return UserHelper.convert_to_user_location(school.location) if school.present? && school.has_attribute?("location")
+
+    lookedup_school = School.where(name: self_reported_school) if self_reported_school.present?
```

Comment:
Maybe lookedup_school could be a memoized method so we can load it only once if we need both the school type and the school location.